.github/workflows/python-publish.yml

```diff
@@ -13,7 +13,8 @@ on:
     types: [published]
 
 permissions:
-  contents: read
+  contents: write
+  id-token: write
```

Comment:
can you please explain this change?

Reply:
id-token is required for signing the artifacts for cosign. 
This github action will sign the artifacts using keyless process, verify signature using bundle file and certificate and sig files. Finally will copy the signing files to the release tag.